packages/kbn-esql-validation-autocomplete/src/autocomplete/autocomplete.ts

```diff
@@ -557,6 +557,7 @@ async function getExpressionSuggestionsByType(
 
   // enrich with assignment has some special rules who are handled somewhere else
   const canHaveAssignments = ['eval', 'stats', 'row'].includes(command.name);
+  const canHaveFunctions = canHaveAssignments || command.name === 'sort';
```

Comment:
Why isn't part of the canHaveAssignments variable?

Reply:
Oof. Good point. I will look into what code the `where` command is hitting...

Reply:
The where command's parameter is of type `boolean` so it hits a different branch. I have updated the code here to be hopefully less confusing (LMK what you think).

However, this whole piece of logic deserves a second look. I find it strange that `canHaveAssignments` is being used to determine whether functions should be shown.

Since this PR is aimed at 8.14 and I don't want to disturb the universe, I will revisit in a follow-up effort.

Reply:
yy sounds fair and thanx for the change, def less confusing!